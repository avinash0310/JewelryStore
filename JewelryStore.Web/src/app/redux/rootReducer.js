import { combineReducers } from 'redux';
import loginReducer from '../../features/Login/reducer';
import jewelryStoreReducer from '../../features/JewelryStore/reducer';

const rootReducer = combineReducers({
    loginDetails: loginReducer,
    jewelryStore: jewelryStoreReducer,
});

export default rootReducer;