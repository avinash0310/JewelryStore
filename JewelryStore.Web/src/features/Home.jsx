import React from 'react';
import { Link } from 'react-router-dom';
import { LOGIN_PAGE_PATH } from '../app/common/constant';

export default class Home extends React.Component {
    render() {
        return (
            <ul className="list-items">
                <li className="item"><Link to={LOGIN_PAGE_PATH}>Login</Link></li>
            </ul>
        );
    };
};