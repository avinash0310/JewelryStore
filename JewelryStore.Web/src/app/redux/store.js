import { applyMiddleware, createStore } from 'redux';
import rootReducer from './rootReducer';
import { httpMiddleware } from '../middleware/middleware.js';

const store = createStore(rootReducer, applyMiddleware(httpMiddleware));
export default store;