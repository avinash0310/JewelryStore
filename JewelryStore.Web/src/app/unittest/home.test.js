import React from 'react';
import { shallow } from "enzyme";
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import Home from '../../features/Home.jsx';
import rootReducer from '../redux/rootReducer';

describe('Home', () => {

    // when
    const store = createStore(rootReducer, applyMiddleware());
    const component = shallow(<Provider store={store}><Home /></Provider>);

    it("should render initial layout", () => {
        // then
        expect(component.getElements()).toMatchSnapshot();
    });

    it("renders list-items", () => {
        // then
        expect(component.find('.list-items')).toBeDefined();
    });

    it("renders items", () => {
        // then
        expect(component.find('.item')).toBeDefined();
    });

    it("renders link", () => {
        // then
        expect(component.find('Link')).toBeDefined();
    });
});