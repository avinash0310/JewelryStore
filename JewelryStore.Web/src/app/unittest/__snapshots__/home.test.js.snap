// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Array [
  <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      }
    }
  >
    <Memo(Connect(Login)) />
  </Context.Provider>,
]
`;

exports[`Home should render initial layout 1`] = `
Array [
  <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      }
    }
  >
    <Home />
  </Context.Provider>,
]
`;

exports[`Home should render initial layout 2`] = `
Array [
  <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      }
    }
  >
    <Home />
  </Context.Provider>,
]
`;

exports[`MyComponent should render initial layout 1`] = `
Array [
  <Context.Provider
    value={
      Object {
        "store": Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "handleChangeWrapper": [Function],
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
      }
    }
  >
    <Memo(Connect(Login)) />
  </Context.Provider>,
]
`;

exports[`MyComponent should render with given state from Redux store 1`] = `
<div>
  <h1>
    Welcome to Jewelry Store
  </h1>
  <label>
    
  </label>
</div>
`;
